pages/create-entry.js

```diff
@@ -17,33 +17,74 @@ import { useState, useEffect } from "react";
 export default function CreateEntry() {
   const router = useRouter();
 
-  const [inputTextFilled, setInputTextFilled] = useState(false);
-  const [categories, setCategories] = useLocalStorage("categories", []);
-  const [listItems, setListItems] = useLocalStorage("listItems", []);
+  //check if list item name has value to activate the button
+  const [submitReady, setSubmitReady] = useState(false);
 
-  useEffect(() => {
-    if (!localStorage.getItem("categories")) {
-      setCategories(JSON.stringify(exampleCategories));
-    }
-  }, []);
+  //variable to check if Input Field with new category has value
+  const [categorySelectionAvailable, setCategorySelectionAvailable] =
+    useState(true);
+
+  const [categories, setCategories] = useLocalStorage(
+    "categories",
+    exampleCategories
+  );
+  const [listItems, setListItems] = useLocalStorage("listItems", []);
+  const [enterInInput, setEnterInInput] = useState(false);
 
   function handleSubmit(event) {
     event.preventDefault();
 
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
 
-    if (data.itemName.length > 2 && data.itemCategory) {
-      setListItems((oldListItems) => [
-        ...oldListItems,
-        { id: nanoid(), name: data.itemName, categoryId: data.itemCategory },
-      ]);
+    //check if user has not pressed enter on input field
+    if (!enterInInput) {
+      if (data.itemName.length > 2) {
+        if (data.newCategory) {
+          let newCategory = { id: nanoid(), name: data.newCategory };
+          setCategories((oldCategories) => [...oldCategories, newCategory]);
+
+          setListItems((oldListItems) => [
+            ...oldListItems,
+            { id: nanoid(), name: data.itemName, categoryId: newCategory.id },
+          ]);
+        } else {
+          setListItems((oldListItems) => [
+            ...oldListItems,
+            {
+              id: nanoid(),
+              name: data.itemName,
+              categoryId: data.itemCategory,
+            },
+          ]);
+        }
+      }
+      router.push(`/`);
+    } else {
+      setEnterInInput(false);
+    }
+  }
+  function handleGoBack(event) {
+    event.preventDefault();
+    if (!enterInInput) {
       router.push(`/`);
+    } else {
+      setEnterInInput(false);
     }
   }
   function handleInput(event) {
     const value = event.target.value;
-    value.length > 2 ? setInputTextFilled(true) : setInputTextFilled(false);
+    value.length > 2 ? setSubmitReady(true) : setSubmitReady(false);
```

Comment:
Can you short this by using the min and max attribute on the input element? just curios

Reply:
nope. Thats not what i need. the `submitReady` is for setting the button disabled or not (to be not able to submit the form)